core/src/main/java/org/apache/calcite/sql/fun/SqlRegexpSplitFunction.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.fun;
+
+import org.apache.calcite.linq4j.Ord;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlUtil;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.type.SqlTypeTransforms;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+/**
+ * Definition of the "REGEXP_SPLIT_TO_ARRAY" builtin SQL function.
+ * The REGEXP_SPLIT_TO_ARRAY(source_string, regexp) split the source string
+ * with the regular expression, returns an array type.
+ */
+public class SqlRegexpSplitFunction extends SqlFunction {
+
+  SqlRegexpSplitFunction() {
+    super(
+        "REGEXP_SPLIT_TO_ARRAY",
+        SqlKind.OTHER_FUNCTION,
+        ReturnTypes.cascade(ReturnTypes.ARG0_NULLABLE_VARYING,
+            SqlTypeTransforms.TO_ARRAY,
+            SqlTypeTransforms.TO_NULLABLE),
+        null,
+        null,
+        SqlFunctionCategory.STRING);
+  }
+
+  @Override public boolean checkOperandTypes(
+      SqlCallBinding callBinding,
+      boolean throwOnFailure) {
+
+    final List<SqlNode> operands = callBinding.operands();
+    assert operands.size() == 2;
+    return OperandTypes.STRING.checkSingleOperandType(callBinding, operands.get(0), 0, true)
+        && OperandTypes.STRING.checkSingleOperandType(callBinding, operands.get(1), 0, true);
```

Comment:
Use `OperandTypes.STRING_STRING` to trigger implicit type coercion.

Reply:
OK. That's better.